Core/src/LongRunning/OperationResponseTrait.php

```diff
@@ -47,17 +47,21 @@ private function operationToArray($operation, $serializer, array $lroMappers)
         $response = $serializer->encodeMessage($response);
 
         $result = null;
-        if ($operation->isDone()) {
-            $type = $response['metadata']['typeUrl'];
+        if ($operation->isDone() && isset($response['response']['typeUrl'])) {
+            $type = $response['response']['typeUrl'];
             $result = $this->deserializeResult($operation, $type, $serializer, $lroMappers);
         }
 
+        $metaType = $response['metadata']['typeUrl'];
```

Comment:
Does this need to be wrapped in an `if` statement similar to line 50?

Reply:
I believe not, the metadata should always be encoded. The metadata will always be there, it will provide progress percentage value when operation is not done yet.